src/App.jsx

```diff
@@ -1,28 +1,91 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+import EmojiPicker from 'emoji-picker-react';
 import './App.css';
 
 const initialColumns = {
   todo: {
     id: 'todo',
     title: 'To Do',
-    tasks: []
-  },
-  inProgress: {
-    id: 'inProgress',
-    title: 'In Progress',
-    tasks: []
-  },
-  done: {
-    id: 'done',
-    title: 'Done',
-    tasks: []
+    tasks: [],
+    order: 0,
+    emoji: '📝',
+    color: '#1a73e8'
   }
 };
 
 function App() {
   const [columns, setColumns] = useState(initialColumns);
   const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [boardTitle, setBoardTitle] = useState('Kanban Board');
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const titleInputRef = useRef(null);
+  const [showEmojiPicker, setShowEmojiPicker] = useState(null);
+  const [showColorPicker, setShowColorPicker] = useState(null);
+  const emojiPickerRef = useRef(null);
+
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
useEffect(() => {
  if (isEditingTitle && titleInputRef.current) {
    titleInputRef.current.focus();
    titleInputRef.current.select();
  }
  return () => {
    if (isEditingTitle) {
      setIsEditingTitle(false);
    }
  };
}, [isEditingTitle]);
```
The useEffect hook for title editing should handle cleanup to prevent memory leaks and potential focus issues. Add a cleanup function to reset the editing state when the component unmounts or dependencies change.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;